apiserver/facades/client/application/application.go

```diff
@@ -270,12 +271,36 @@ func newFacadeBase(ctx facade.Context) (*APIBase, error) {
 		return nil, errors.Trace(err)
 	}
 
+	state := &stateShim{ctx.State()}
+
+	modelCfg, err := model.Config()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var chCfg charmhub.Config
+	chURL, ok := modelCfg.CharmHubURL()
+	if ok {
+		chCfg, err = charmhub.CharmHubConfigFromURL(chURL, logger.Child("client"))
+	} else {
+		chCfg, err = charmhub.CharmHubConfig(logger.Child("client"))
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	chClient, err := charmhub.NewClient(chCfg)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	updateSeries := NewUpdateSeriesAPI(state, makeUpdateSeriesValidator(chClient))
+
 	return NewAPIBase(
-		&stateShim{ctx.State()},
+		state,
 		storageAccess,
 		ctx.Auth(),
+		updateSeries,
```

Comment:
I wonder what difference it would make to have lazy instantiation of these "sub"-APIs across the board.

Reply:
Yeah, let's think about this, it seems like this eager/strict construction isn't ideal. For one controller it could hold onto MANY clients.